### PR TITLE
Bone Orientation + World Orientation Fix, Anim Import + Export

### PR DESCRIPTION
**Pull request by ssbucarlos**
Starting the pull request early to plan the merge. Due to the nature of modifying the bone orientation + world orientation of meshes of import, this means that mesh export needs to be adjusted and tested. In addition, models imported will no longer be compatible with the previous .nuanmb importer and exporter script. As such i dont want to merge this until the new animation workflow is complete. (Also im not too familiar with like project planning stuff so if an early pull request is not the way to go for stuff like this let me know)

- [x] Skel Import
- [x] Skel Export
- [x] Mesh Import
- [x] Mesh Export
- [x] Anim Import
- [ ] Anim Export
- [ ] Test Fighters
- [ ] Test Stages
- [ ] Test Cameras

### CODE REVIEW

**Comment by ScanMountGoat** [comment]
Scale compensation and scale inheritance are independent, so there's no need to handle four cases of true/true, false/true, etc. We should be able to just bake scale compensation. I'll try and take a look at implementing it later. Is there a reason these are set as custom properties on the blender bones?

**Comment by ssbucarlos** [comment]
Im honestly still not 100% sure how to really handle compensate_scale. I found that to best replicate sonics scaling anims (in particular i used jumpf and fsmash as reference) i needed the different logic for when compensate_scale was true and inherit_scale was false. If i remember correctly, using the formula for all cases resulted in incorrect transforms elsewhere. And none of blenders available scaling options was able to give the correct preview. As for the custom properties being setup, the plan was to originally store these to setup a driver on the scale inheritance value for the blender bone, and drive the correct option based on these values. However, the other scaling types ended up going unused, so i might just end up removing the custom properties + drivers.

**Comment by ScanMountGoat** [comment]
I'm not 100% confident on the new scale code working all the time, but matching Cross Mod is probably good enough for now. I'm not sure how you want to handle key frames for compensate_scale, so I'll leave that up to you.

**Comment by ssbucarlos** [comment]
ty for looking into this, ill pull the changes and test when i get the chance. I never thought about setting the bone scale directly instead of dealing with the matrix haha. This section of the code is now significantly cleaner and well explained, tyvm

File: panels/import_model.py [hunk 1]
```diff
@@ -671,4 +827,124 @@ def setup_blender_mat(blender_mat, material_label, ssbh_matl: ssbh_data_py.matl_
...
+    bpy.ops.object.mode_set(mode='POSE', toggle=False)
+    owner_bone = armature.pose.bones.get(owner_bone_name, None)
+    if owner_bone is not None:
+        x,y,z = 'X', 'Y', 'Z'
+        for index, axis in enumerate([x,y,z]):
+            crc = owner_bone.constraints.new('COPY_ROTATION')
+            crc.name = f'{constraint_name}.{axis}'
+            crc.target = armature
+            crc.subtarget =  target_bone_name
+            crc.target_space = 'LOCAL_OWNER_ORIENT'
```
Comment (ScanMountGoat) [inline panels/import_model.py:944]: This breaks compatibility with Blender 2.93. This option was added for 3.0. https://docs.blender.org/api/2.93/bpy.types.Constraint.html#bpy.types.Constraint.target_space

**Comment by ScanMountGoat** [comment]
For animation export, `track.scale_options.compensate_scale` should always be `False`. The scale compensation is "baked" on import, so there's no need to enable it again on export. The scale inheritance still needs to be set properly based on the scale inheritance used in Blender. Thankfully, it should be pretty easy to check in Cross Mod or in game if the scale exporter isn't working as expected.

**Comment by ScanMountGoat** [comment]
@ssbucarlos What are your thoughts on merging this? We can always move animation export into a separate branch/PR. If you've made a lot of progress on export, it makes sense to just wait. The main thing blocking the merge would be if any of the existing features are broken/missing. The advantage to merging earlier is fewer conflicts later since fixing some of the current issues may conflict with the changes in this branch.

**Comment by ssbucarlos** [comment]
Good idea, but lets wait a week to see if i can make some time to finish up export, and if not lets just go ahead and merge so it doesn't become significantly more difficult to merge later.

File: panels/import_model.py (hunk 1 above)
Comment (ssbucarlos) [reply panels/import_model.py:944]: oh yeah i think i misunderstood local_owner_orientation anyways. im not really sure how to get consistent interpolation type helper bone entry behavior, going to try to make it work with local space and that will maintain pre 3.0 support.

**Comment by ScanMountGoat** [comment]
Sounds good. Thanks for the update. ~~now I can procrastinate longer on updating ssbh_data_py~~

**Comment by ScanMountGoat** [comment]
Apparently github really wants this PR closed lol.

**Comment by ScanMountGoat** [comment]
Merging the commits forces this PR to remain closed, so this discussion has now moved to https://github.com/ssbucarlos/smash-ultimate-blender/pull/91

**Pull request merged**